### PR TITLE
Feature/ios performances

### PR DESCRIPTION
**Pull request by matteocrippa**
- iOS fixes background animation error on heading (old patch sent to official repo but never accepted)
- port fix for snapshot crash

### CODE REVIEW

**Comment by petr-pokorny-1** [comment]
Thank you!

I am wondering - do files under platform/ios/platform/ios/Mapbox.playground/build/ have to be checked in?

File: .gitignore [hunk 1]
```diff
@@ -16,6 +16,8 @@
 /platform/android/build
 /platform/android/MapboxGLAndroidSDK/src/main/assets/sdk_versions/com.mapbox.mapboxsdk
 *.code-workspace
+
```
Comment (roblabs) [inline .gitignore:19]: Does Mapbox.playground allow for adding files to Git ignore?

That is, is it possible to remove the Playground binary build files by adding the following?

```
platform/ios/platform/ios/Mapbox.playground/build/
```

**Comment by petr-pokorny-1** [comment]
Looking good. @matteocrippa do you have any steps to follow to reproduce the crash?

File: .gitignore (hunk 1 above)
Comment (matteocrippa) [reply .gitignore:20]: Should be fine now

**Comment by matteocrippa** [comment]
> Looking good. @matteocrippa do you have any steps to follow to reproduce the crash?

Which crash? 🐙

**Comment by petr-pokorny-1** [comment]
> > Looking good. @matteocrippa do you have any steps to follow to reproduce the crash?
> 
> Which crash? 🐙

You mentioned in this PR "port fix for snapshot crash" - so i though I would test it manually on device if possible.

**Comment by matteocrippa** [comment]
Yeah, sorry (long day)... yeah it's the issue 485 of the mapbox repo.
I just port the patch from there.
As a user I was experiencing that randomly, but was pretty consistent issue in crash log for device with iOS 14.x

**Comment by petr-pokorny-1** [comment]
Got it, thanks! Related issue: https://github.com/mapbox/mapbox-gl-native-ios/issues/485

**Review by petr-pokorny-1 — APPROVED** [review]

**Pull request merged**